### PR TITLE
Use localized format for idle-screen/gui

### PR DESCRIPTION
**Pull request by domcross**
#### Description
GUI/idle-screen shows weekday name and date in localized format.

#### Type of PR
- [x] Bugfix
- [x] Feature implementation
- [ ] Refactor of code (without functional changes)
- [ ] Documentation improvements
- [ ] Test improvements

#### CLA
- [x] CLA signed

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 import mycroft.audio
 from adapt.intent import IntentBuilder
+from lingua_franca.format import date_time_format
```
Comment (krisgesling) [inline __init__.py]: Would be safer to import this from `mycroft.util.format`

As Lingua Franca becomes more useful for broader projects, we have been including some compatibility code in mycroft-core to ensure it keeps playing nicely.

File: __init__.py [hunk 2]
```diff
@@ -622,12 +623,26 @@ def show_date_mark1(self, location, day):
...
-        return day.strftime("%B %d")
+        month = nice_date(day)  # forces lingua_franca to initialize
+        if self.lang in date_time_format.lang_config.keys():
+            month = date_time_format.lang_config[self.lang]['month'][str(int(day.strftime("%m")))]
+        else:
+            month = day.strftime("%B")
+        if self.config_core.get('date_format') == 'MDY':
+            return "{} {}".format(month, day.strftime("%d"))
+        else:
+            return "{}. {}".format(day.strftime("%d"), month)
```
Comment (krisgesling) [inline __init__.py]: Is there period here intentional? I'm guessing that is a further piece of localization?

In Australia we use DMY, but generally no .

File: __init__.py [hunk 3]
```diff
@@ -622,12 +623,26 @@ def show_date_mark1(self, location, day):
     def get_weekday(self, day=None, location=None):
         if not day:
             day = self.get_local_datetime(location)
-        return day.strftime("%A")
+        weekday = nice_date(day)  # forces lingua_franca to initialize
```
Comment (krisgesling) [inline __init__.py]: Shouldn't need these if we import from `mycroft.util`

File: __init__.py [hunk 4]
```diff
@@ -622,12 +623,26 @@ def show_date_mark1(self, location, day):
...
+        if self.lang in date_time_format.lang_config.keys():
+             weekday = list(
+                 date_time_format.lang_config[self.lang]['weekday'].values())[day.weekday()]
+        else:
+            weekday = day.strftime("%A")
+        return weekday
```
Comment (krisgesling) [inline __init__.py]: It seems these are returned all lowercase, so would need to add a `.capitalize()` for the weekday and month.

File: __init__.py [hunk 5]
```diff
@@ -622,12 +623,26 @@ def show_date_mark1(self, location, day):
     def get_weekday(self, day=None, location=None):
         if not day:
             day = self.get_local_datetime(location)
-        return day.strftime("%A")
+        weekday = nice_date(day)  # forces lingua_franca to initialize
+        if self.lang in date_time_format.lang_config.keys():
+             weekday = list(
+                 date_time_format.lang_config[self.lang]['weekday'].values())[day.weekday()]
```
Comment (krisgesling) [inline __init__.py]: This is all correct but it's quite a mouthful, I wonder if we can split it to make it clearer for people reading the code in the future. Maybe something like:
```
localized_day_names = list(date_time_format.lang_config[self.lang]['weekday'].values())
weekday = localized_day_names[day.weekday()]
```

**Review by krisgesling — COMMENTED** [review]
Hey Dom, thanks for this. Definitely a good change to make.

I haven't tested it across languages but it's working for English and I'm sure you tested it for German. 

I had a few suggestions and questions:

File: __init__.py (hunk 2 above)
Comment (domcross) [reply __init__.py]: yes, please excuse my ignorance here.

File: __init__.py (hunk 3 above)
Comment (domcross) [reply __init__.py]: Thanks, this is much better.

File: __init__.py (hunk 5 above)
Comment (domcross) [reply __init__.py]: In German weekday names are always capitalized, so again I assumed something which isn't always valid...

File: __init__.py (hunk 3 above)
Comment (domcross) [reply __init__.py]: I have a another PR https://github.com/MycroftAI/skill-weather/pull/156 and as soon as I removed this "lingua_franca initializiation" there as well the localization does no longer work :frowning_face: 
Will keep the initialization until a better idea shows up...

Comment (domcross) [reply __init__.py]: Could not find any workaround for initialization of date_time_format, but calling `date_time_format.cache(self.lang)` during skill initialize is more elegant...

Comment (krisgesling) [reply __init__.py]: yeah I just had a quick look and couldn't find anything better :+1:

**Comment by krisgesling** [comment]
Thanks for doing the updates. 

I made a tiny addition to ensure the month is always capitalized also.

**Pull request merged**